tico/passes/cast_clamp_mixed_type_args.py

```diff
@@ -0,0 +1,147 @@
+# Copyright (c) 2025 Samsung Electronics Co., Ltd. All Rights Reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    import torch.fx
+import torch
+from torch.export import ExportedProgram
+
+from tico.utils import logging
+from tico.utils.graph import create_node
+from tico.utils.passes import PassBase, PassResult
+from tico.utils.trace_decorators import trace_graph_diff_on_pass
+from tico.utils.utils import is_target_node, set_new_meta_val
+from tico.utils.validate_args_kwargs import ClampArgs
+
+
+@trace_graph_diff_on_pass
+class CastClampMixedTypeArgs(PassBase):
+    """
+    This pass ensures consistent dtypes for clamp operations by:
+    1. Converting min/max arguments to match output dtype when provided
+    2. Inserting cast operations when input dtype differs from output dtype
+
+    Behavior Examples:
+    - When input dtype differs from output:
+        Inserts _to_copy operation to convert input
+    - When min/max dtype differs from output:
+        Converts min/max values to output dtype
+
+    [before, if input dtype is different from output dtype]
+
+            input               min(or max)
+           (dtype=int)         (dtype=float)
+              |                    |
+            clamp <----------------+
+              |
+            output
+           (dtype=float)
+
+    [after]
+
+            input             min(or max)
+           (dtype=int)       (dtype=float)
+              |                  |
+            cast                 |
+          (in=int, out=float)    |
+              |                  |
+            clamp <--------------+
+              |
+            output
+           (dtype=float)
+    """
+
+    def __init__(self):
+        super().__init__()
+
+    def convert(self, exported_program: ExportedProgram, node: torch.fx.Node) -> bool:
+        logger = logging.getLogger(__name__)
+        modified = False
+
+        graph_module = exported_program.graph_module
+        graph = graph_module.graph
+
+        # clamp(Tensor self, Scalar? min=None, Scalar? max=None) -> Tensor
+        args = ClampArgs(*node.args, **node.kwargs)  # type: ignore[arg-type]
+
+        input = args.input
+        min = args.min
+        max = args.max
+
+        if "val" not in input.meta or "val" not in node.meta:
+            logger.warning(f"Missing meta['val'] for node {node.name}")
+            return False
```

Comment:
Just curiosity, is there any case that meets this condition? AFAIK, all fx nodes have `meta["val"]` at this point.